lightningd/plugin.c

```diff
@@ -675,8 +682,21 @@ static bool plugin_rpcmethod_add(struct plugin *plugin,
 		return false;
 	}
 
+
 	cmd = notleak(tal(plugin, struct json_command));
 	cmd->name = json_strdup(cmd, buffer, nametok);
+
+	if (argstok) {
+		char *ret = tal_strdup(cmd, "");
+		for (size_t i = 0; i < argstok->size; i++) {
+			ret = tal_strcat(cmd, take(ret), json_strdup(cmd, buffer, json_get_arr(argstok, i)));
+			ret = tal_strcat(cmd, take(ret), " ");
+		}
+		cmd->usage = tal_strdup(cmd, ret);
```

Comment:
You can use `tal_steal` to make `cmd` the owner without copying. That'd also address the memory leak resulting from `ret` not being freed.